src/Main.js

```diff
@@ -1,15 +1,37 @@
 import React from 'react';
 import HornedBeast from './HornedBeast.js';
+import Form from 'react-bootstrap/Form';
+import data from './data.json';
 
 import './Main.css'
+
 class Main extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      beasts: data
+    }
+  }
+
+  filter = (e) => {
+    let numHorns = parseInt(e.target.value);
+    let displayedBeast = data;
+    if (numHorns) {
+      displayedBeast = data.filter((creature) => creature.horns === numHorns);
+    }
+    this.setState({ beasts: displayedBeast });
+    console.log(this.state.beasts)
+
+
+  }
```

Comment:
- If we did end up passing data as props from app, then we could approach this a different way by setting horns as 0 for the initial state value and then in our filter set the state of the horns to be the` e.target.value`

There are always ways we can simplify a bit 👍